robot/rospackages/src/task_handler/scripts/task_handler_server.py

```diff
@@ -106,10 +161,50 @@ def handle_task(task, status, args):
     return response + "\n"
 
 def task_handler_server():
+    global local
     rospy.init_node('task_handler_server')
     s = rospy.Service('task_handler', HandleTask, handle_task_request)
-    print("Ready to respond to task handling requests")
+    ready_msg = "Ready to respond to task handling requests"
+
+    if local:
+        ready_msg += " (local mode)"
+
+    print(ready_msg)
     rospy.spin()
 
 if __name__ == "__main__":
+    local = False
+
+    if "local" in sys.argv:
+        print("Task handler local mode supported tasks")
+        print("--> camera_stream")
+        local = True
+
+    # this value is set at runtime based off how many video port devices are identified
+    max_streams = 0
+    # how many streams are currently on
+    stream_ctr = 0
+
+    current_dir = os.path.dirname(os.path.realpath(__file__)) + "/"
+    mcu_control_dir = current_dir + '../../mcu_control/scripts/'
+
+    # tasks that can be run
+    scripts = [mcu_control_dir + "RoverNode.py", mcu_control_dir + "ArmNode.py", mcu_control_dir + "ScienceNode.py", mcu_control_dir + "PdsNode.py", current_dir + "start_ros_stream.sh"]
+
+    # set up listener objects
+    running_tasks = [Listener(scripts[0], "python3"), Listener(scripts[1], "python3"), Listener(scripts[2], "python3"), Listener(scripts[3], "python3"), Listener(scripts[4], "bash", "", 1, True)]
+
+    # expected client arguments for choosing task
+    known_tasks = ["rover_listener", "arm_listener", "science_listener", "pds_listener", "camera_stream"]
+    known_listeners = known_tasks[:-1]
+
+    # keep track of currently running streams
+    active_ports = []
+    active_streams = []
+
+    i = 0
+    for task in running_tasks:
+        task.set_name(known_tasks[i])
+        i += 1
```

Comment:
for i in range(len(running_tasks)):
    task = running_tasks[i]
    task.............

This makes sure that the scope of the iteration variable is correct

Reply:
Fixed, thanks